OSDOCS-10713: Documented the 4.15.15 z-stream release notes (#76653)
openshift/openshift-docs: release_notes/ocp-4-15-release-notes.adoc

For any {product-title} release, always review the instructions on xref:../updating/updating_a_cluster/updating-cluster-web-console.adoc#updating-cluster-web-console[updating your cluster] properly.
====

// 4.15.15
[id="ocp-4-15-15"]
=== RHSA-2024:3327 - {product-title} 4.15.15 bug fix and security update

Issued: 2024-05-29

{product-title} release 4.15.15, which includes security updates, is now available. The list of bug fixes that are included in the update is documented in the link:https://access.redhat.com/errata/RHSA-2024:3327[RHSA-2024:3327] advisory. The RPM packages that are included in the update are provided by the link:https://access.redhat.com/errata/RHBA-2024:3332[RHBA-2024:3332] advisory.

Space precluded documenting all of the container images for this release in the advisory.

You can view the container images in this release by running the following command:

[source,terminal]
----
$ oc adm release info 4.15.15 --pullspecs
----

[id="ocp-4-15-15-bug-fixes"]
==== Bug fixes

> * Previously, certain situations caused transfer of an Egress IP address from one node to a different node to fail, and this failure impacted the OVN-Kubernetes network. The network failed to send gratuitous Address Resolution Protocol (ARP) requests to peers to inform them of the new node's medium access control (MAC) address. As a reult, peers would temporarily send reply traffic to an old node and this traffic led to failover issues. With this release, the OVN-kubernetes network correctly sends a gratuitous ARP to peers to inform them of the new Egress IP node MAC address, so that each peer can send reply traffic to the new node without causing failover time issues. (link:https://issues.redhat.com/browse/OCPBUGS-33960[*OCPBUGS-33960*]

Reviewer: 3 bits:
```suggestion
* Previously, certain situations caused transfer of an Egress IP address from one node to a different node to fail, and this failure impacted the OVN-Kubernetes network. The network failed to send gratuitous Address Resolution Protocol (ARP) requests to peers to inform them of the new node's medium access control (MAC) address. As a result, peers would temporarily send reply traffic to an old node and this traffic led to failover issues. With this release, the OVN-Kubernetes network correctly sends a gratuitous ARP to peers to inform them of the new Egress IP node MAC address, so that each peer can send reply traffic to the new node without causing failover time issues. (link:https://issues.redhat.com/browse/OCPBUGS-33960[*OCPBUGS-33960*])
```

> * Previously, when mirroring operator catalogs, the `oc-mirror` rebuilt the catalogs and regenerated the catalogs internal cache according to the `imagesetconfig` catalog filtering specifications. This operation required the use of the `opm` binary found within the catalogs. In {product-title} 4.15, operator catalogs include the `opm` {op-system-base-full} 9 binary and this caused the mirroring process to fail when running on {op-system-base} 8 systems. With this release, `oc-mirror` no longer builds catalogs by default. Instead, catalogs are mirrored directly to their destination registries. (link:https://issues.redhat.com/browse/OCPBUGS-33575[*OCPBUGS-33575*]

Reviewer: Note ending `)` here too:
```suggestion
* Previously, when mirroring Operator catalogs, the `oc-mirror` CLI plugin rebuilt the catalogs and regenerated the catalogs' internal cache according to the `imagesetconfig` catalog filtering specifications. This operation required the use of the `opm` binary found within the catalogs. In {product-title} 4.15, Operator catalogs include the `opm` {op-system-base-full} 9 binary, and this caused the mirroring process to fail when running on {op-system-base} 8 systems. With this release, `oc-mirror` no longer builds catalogs by default. Instead, catalogs are mirrored directly to their destination registries. (link:https://issues.redhat.com/browse/OCPBUGS-33575[*OCPBUGS-33575*])
```

* Previously, the image registry did not support Amazon Web Services (AWS) region `ca-west-1`. With this release, the image registry can now be deployed in this region. (link:https://issues.redhat.com/browse/OCPBUGS-33672[*OCPBUGS-33672*])

> * Previously, ServiceAccounts (SAs) could not be used as OAuth2 clients because there were no tokens associated with the SAs. With this release, the OAuth registry client has been modified to anticipate this case and the issue has been resolved. (link:https://issues.redhat.com/browse/OCPBUGS-33210[*OCPBUGS-33210*])

Reviewer: Either:
```suggestion
* Previously, `ServiceAccount` (SA) objects could not be used as OAuth2 clients because there were no tokens associated with the SAs. With this release, the OAuth registry client has been modified to anticipate this case and the issue has been resolved. (link:https://issues.redhat.com/browse/OCPBUGS-33210[*OCPBUGS-33210*])
```
or:
```suggestion
* Previously, service accounts (SAs) could not be used as OAuth2 clients because there were no tokens associated with the SAs. With this release, the OAuth registry client has been modified to anticipate this case and the issue has been resolved. (link:https://issues.redhat.com/browse/OCPBUGS-33210[*OCPBUGS-33210*])
```

> * Previously, the proxy information set in the install config was not applied to the bootstrap process. With this release, the proxy information is applied to the bootstrap ignition data which is applied to the bootstrap machine and the issue has been resolved. (link:https://issues.redhat.com/browse/OCPBUGS-33205[*OCPBUGS-33205*])

Reviewer: Capitalize "Ignition":
```suggestion
* Previously, the proxy information set in the install config was not applied to the bootstrap process. With this release, the proxy information is applied to the bootstrap Ignition data which is applied to the bootstrap machine and the issue has been resolved. (link:https://issues.redhat.com/browse/OCPBUGS-33205[*OCPBUGS-33205*])
```
I think also (from looking at the original, pre-clone [Jira](https://issues.redhat.com/browse/OCPBUGS-12890)) that this would more accurately (and align with how the rest of the repo refers to it) be:
```suggestion
* Previously, the proxy information set in the `install-config.yaml` file was not applied to the bootstrap process. With this release, the proxy information is applied to the bootstrap Ignition data which is applied to the bootstrap machine and the issue has been resolved. (link:https://issues.redhat.com/browse/OCPBUGS-33205[*OCPBUGS-33205*])
```
Or if you'd rather be a little more generic just to be safe, at least add "file":
```suggestion
* Previously, the proxy information set in the install config file was not applied to the bootstrap process. With this release, the proxy information is applied to the bootstrap Ignition data which is applied to the bootstrap machine and the issue has been resolved. (link:https://issues.redhat.com/browse/OCPBUGS-33205[*OCPBUGS-33205*])
```

> * Previously, the proxy information set in the install config was not applied to the bootstrap process. With this release, the proxy information is applied to the bootstrap ignition data which is applied to the bootstrap machine and the issue has been resolved. (link:https://issues.redhat.com/browse/OCPBUGS-33205[*OCPBUGS-33205*])

Reviewer: Looks like a dupe:
```suggestion
```

> * Previously, the information from `imageRegistryOverrides` was only extracted once on the HyperShift Operator initialization and did not refresh. With this release, the Hypershift Operator retrieves the new `ImageContentSourcePolicy` files from the management cluster and adds them to the Hypershift Operator and Control Plane Operator in every reconciliation loop. (link:https://issues.redhat.com/browse/OCPBUGS-33117[*OCPBUGS-33117*])

Reviewer: ```suggestion
* Previously, the information from the `imageRegistryOverrides` setting was only extracted once on the HyperShift Operator initialization and did not refresh. With this release, the Hypershift Operator retrieves the new `ImageContentSourcePolicy` files from the management cluster and adds them to the Hypershift Operator and Control Plane Operator in every reconciliation loop. (link:https://issues.redhat.com/browse/OCPBUGS-33117[*OCPBUGS-33117*])
```

* Previously, the Hypershift Operator was not using the `RegistryOverrides` mechanism to inspect the image from the internal registry. With this release, the metadata inspector works as expected during the Hypershift Operator reconciliation, and the `OverrideImages` are properly populated. (link:https://issues.redhat.com/browse/OCPBUGS-32220[*OCPBUGS-32220*])

> * Previously, attempting to update the OpenShift vSphere connection configuration failed if the configuration included `””` characters. With this release, the characters are stored correctly and the issue has been resolved. (link:https://issues.redhat.com/browse/OCPBUGS-31863[*OCPBUGS-31863*])

Reviewer: Wonder if this could more specifically be:
```suggestion
* Previously, attempting to update the {vmw-first} connection configuration for {product-title} failed if the configuration included `””` characters. With this release, the characters are stored correctly and the issue has been resolved. (link:https://issues.redhat.com/browse/OCPBUGS-31863[*OCPBUGS-31863*])
```

[id="ocp-4-15-15-known-issues"]
==== Known issue

> * Previously, after upgrading to 4.15.6, attempting to use `oc-mirror` within a cluster failed. With this release, there is now a FIPS-compliant version of `oc-mirror` for RHEL 9 and a version of `oc-mirror` for RHEL 8 that is not FIPS-compliant. (link:https://issues.redhat.com/browse/OCPBUGS-31609[*OCPBUGS-31609*])

Reviewer: ```suggestion
* Previously, after upgrading to {product-title} 4.15.6, attempting to use the `oc-mirror` CLI plugin within a cluster failed. With this release, there is now a FIPS-compliant version of `oc-mirror` for RHEL 9 and a version of `oc-mirror` for RHEL 8 that is not FIPS-compliant. (link:https://issues.redhat.com/browse/OCPBUGS-31609[*OCPBUGS-31609*])
```
